examples/raw-template/raw_template.bal

```diff
@@ -0,0 +1,63 @@
+import ballerina/lang.'object;
+import ballerina/io;
+
+public function main() {
+    string name = "Ballerina";
+    // Create a raw template with the `name` variable interpolated in the
+    // literal.
+    'object:RawTemplate template = `Hello ${name}!!!`;
+    // The string literal portions of the raw template are accessible
+    // through the `strings` field. The `strings` field is a list of
+    // `string` values. The string literal portions of the raw template
+    // are separated by the interpolations. If there are `n` insertions,
+    // there will be `n+1` strings.
+    io:println(template.strings);
+    // The `insertions` field holds the values obtained by evaluating
+    // the expressions specified in the interpolations of the raw template.
+    // In this case, there is just one interpolation: a reference to the
+    // variable, `name`.
+    io:println(template.insertions[0]);
+
+    Person[] people = [
+        {name: "John Doe", age: 25},
+        {name: "Jane Doe", age: 20}
+    ];
+
+    Query[] queries = [];
+    int i = 0;
+
+    foreach var p in people {
+        queries[i] = (`INSERT INTO Details VALUES (${p.name}, ${p.age})`);
+        i += 1;
+    }
```

Comment:
Shall we do 

```suggestion

    foreach var p in people {
        queries.push(`INSERT INTO Details VALUES (${p.name}, ${p.age})`);
    }
```

instead?

Reply:
This is what I initially tried to do. But ran in to this: https://github.com/ballerina-platform/ballerina-lang/issues/24268 :sweat_smile:

Reply:
:sweat_smile: